src/main/resources/db/changelog/20240329143555-add_environment_content_override_schema.xml

```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.19.xsd">
+
+    <changeSet id="20240329143555-1" author="crog">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <columnExists tableName="cp_content_override" columnName="environment_id"/>
+            </not>
+        </preConditions>
+
+        <addColumn tableName="cp_content_override">
+            <column name="environment_id" type="VARCHAR(32)">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+
+        <addForeignKeyConstraint baseTableName="cp_content_override"
+            baseColumnNames="environment_id"
+            referencedTableName="cp_environment"
+            referencedColumnNames="id"
+            constraintName="fk_content_override_environment"
+            onDelete="CASCADE" />
+    </changeSet>
+
+    <changeSet id="20240329143555-2" author="crog">
+        <!--
+            Scrub the data to ensure we don't have malformed content overrides with both an AK and
+            a consumer set. We'll follow the design's outlined priority and clear the consumer on
+            any row that has a non-null key ID.
+
+            This way we need not worry (as much) about the constraint we'll be adding in the next
+            task.
+        -->
+
+        <sql dbms="mariadb, postgresql">
+            UPDATE cp_content_override SET consumer_id = NULL WHERE key_id IS NOT NULL
```

Comment:
Looking at this during self-review and I'm beginning to wonder if it'd be better/safer to clear the consumer instead. Both options kinda suck here for the same reason:

- If we clear the activation key, existing consumers that are currently getting the affected override will continue getting it, but new consumers registering with the activation key will no longer inherit the override.
- If we clear the consumer as we do here, existing consumers that may have registered with the activation key to get the override will no longer get it, but new consumers registering with it will.

In either case we end up in a weird state desync that will be confusing. But failing to do either will cause the addition of the check constraint to fail.

Reply:
are there actually any such data in prod?

Reply:
I just run this query and it seems there aren't (in hosted at least)

Reply:
In prod, no; but we don't know what the many DBs of Sat installs look like, though. And if there are none, we're in a great place. The concern is that for some reason we never required the parent object on our content override subtypes, nor did we have a constraint preventing two parents, so it's very possible something got busted.

Reply:
I can't see how this kind of malformed data would appear through our APIs